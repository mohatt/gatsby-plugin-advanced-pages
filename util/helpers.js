const path = require('path')
const debug = require("debug")("gatsby-plugin-advanced-pages")

// Checks if a path is a child of another one
function isPathChildof (child, parent) {
  const relative = path.relative(parent, child)
  return relative && !relative.startsWith('..') && !path.isAbsolute(relative)
}

module.exports = {
  debug,
  isPathChildof,
}
