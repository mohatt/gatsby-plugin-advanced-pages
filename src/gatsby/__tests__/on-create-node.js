import { onCreateNode } from '../'
import { mountOptions } from '../../../test/helpers'
import testCases from '../../../test/__fixtures__/on-create-node'

// Use in-memory file system
jest.mock('fs')

// Mount default options
beforeAll(() => { 
  mountOptions()
})

describe(`onCreateNode`, () => {
  const createNode = jest.fn()
  const createParentChildLink = jest.fn()
  const actions = { createNode, createParentChildLink }

  const getNode = jest.fn()
  const createNodeId = jest.fn().mockImplementation((v) => ('id-' + v).toUpperCase())
  const createContentDigest = jest.fn().mockImplementation((v) => JSON.stringify(v))

  for (const testCase of testCases) {
    it(`correctly creates a Page node from a markdown node (${testCase.title})`, () => {
      getNode.mockReturnValue(testCase.parent)
      onCreateNode({
        node: testCase.node,
        actions,
        getNode,
        createNodeId,
        createContentDigest
      })
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(createParentChildLink.mock.calls).toMatchSnapshot()

      createNode.mockClear()
      createParentChildLink.mockClear()
    })
  }
})
