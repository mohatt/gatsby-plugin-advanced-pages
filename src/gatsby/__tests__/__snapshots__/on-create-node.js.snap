// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onCreateNode correctly creates a Page node from a markdown node (testCase1) 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "id": "ID-/WHATEVER/PATH",
      "internal": Object {
        "contentDigest": "{\\"id\\":\\"ID-/WHATEVER/PATH\\",\\"name\\":\\"whatever\\",\\"path\\":\\"/whatever/path\\",\\"parent\\":\\"ID-WHATEVER >>> PAGE\\",\\"children\\":[],\\"internal\\":{\\"type\\":\\"Route\\",\\"description\\":\\"Advanced Pages (Route): /whatever/path\\"}}",
        "description": "Advanced Pages (Route): /whatever/path",
        "type": "Route",
      },
      "name": "whatever",
      "parent": "ID-WHATEVER >>> PAGE",
      "path": "/whatever/path",
    },
  ],
  Array [
    Object {
      "children": Array [
        "ID-/WHATEVER/PATH",
      ],
      "data": "whatever",
      "helper": "whatever",
      "id": "ID-WHATEVER >>> PAGE",
      "internal": Object {
        "contentDigest": "{\\"id\\":\\"ID-WHATEVER >>> PAGE\\",\\"title\\":\\"whatever\\",\\"template\\":\\"whatever\\",\\"helper\\":\\"whatever\\",\\"data\\":\\"whatever\\",\\"parent\\":\\"whatever\\",\\"children\\":[\\"ID-/WHATEVER/PATH\\"],\\"internal\\":{\\"type\\":\\"Page\\",\\"description\\":\\"Advanced Pages (Page): whatever\\"}}",
        "description": "Advanced Pages (Page): whatever",
        "type": "Page",
      },
      "parent": "whatever",
      "template": "whatever",
      "title": "whatever",
    },
  ],
]
`;

exports[`onCreateNode correctly creates a Page node from a markdown node (testCase1) 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "children": Array [
          "ID-/WHATEVER/PATH",
        ],
        "data": "whatever",
        "helper": "whatever",
        "id": "ID-WHATEVER >>> PAGE",
        "internal": Object {
          "contentDigest": "{\\"id\\":\\"ID-WHATEVER >>> PAGE\\",\\"title\\":\\"whatever\\",\\"template\\":\\"whatever\\",\\"helper\\":\\"whatever\\",\\"data\\":\\"whatever\\",\\"parent\\":\\"whatever\\",\\"children\\":[\\"ID-/WHATEVER/PATH\\"],\\"internal\\":{\\"type\\":\\"Page\\",\\"description\\":\\"Advanced Pages (Page): whatever\\"}}",
          "description": "Advanced Pages (Page): whatever",
          "type": "Page",
        },
        "parent": "whatever",
        "template": "whatever",
        "title": "whatever",
      },
      "parent": Object {
        "frontmatter": Object {
          "data": "whatever",
          "helper": "whatever",
          "routes": Object {
            "whatever": "/whatever/path",
          },
          "template": "whatever",
          "title": "whatever",
        },
        "id": "whatever",
        "internal": Object {
          "type": "MarkdownRemark",
        },
        "parent": "whatever",
      },
    },
  ],
]
`;
